core/src/test/scala/org/apache/spark/sql/delta/schema/CheckConstraintsSuite.scala

```diff
@@ -367,4 +367,48 @@ class CheckConstraintsSuite extends QueryTest
     }
   }
 
+  // TODO fix this
+  testQuietly("SET NOT NULL constraint fails") {
+    withSQLConf((DeltaSQLConf.DELTA_PROTOCOL_DEFAULT_WRITER_VERSION.key, "3")) {
+      withTable("my_table") {
+        sql("CREATE TABLE my_table (id INT) USING DELTA;")
+        sql("INSERT INTO my_table VALUES (1);")
+        val e = intercept[AnalysisException] {
+          sql("ALTER TABLE my_table CHANGE COLUMN id SET NOT NULL;")
+        }.getMessage()
+        assert(e.contains("Cannot change nullable column to non-nullable"))
+      }
+
+    }
+  }
+
+  // TODO fix this
+  testQuietly("ending semi-colons make ADD, DROP constraint commands fail") {
+    withTable("my_table") {
+      sql("CREATE TABLE my_table (birthday DATE) USING DELTA;")
+      sql("INSERT INTO my_table VALUES ('2021-11-11');")
+
+      // ADD - no semi-colon will PASS
+      sql("ALTER TABLE my_table ADD CONSTRAINT aaa CHECK (birthday > '1900-01-01')")
+      sql("ALTER TABLE my_table ADD CONSTRAINT bbb CHECK (birthday > '1900-02-02')")
+
+      // ADD - with a semi-colon will FAIL
+      val e = intercept[ParseException] {
+        sql("ALTER TABLE my_table ADD CONSTRAINT ccc CHECK (birthday > '1900-03-03');")
+      }.getMessage()
+      assert(e.contains("no viable alternative at input 'ALTER TABLE my_table ADD CONSTRAINT'" +
+        "(line 1, pos 25)"))
+
+      // DROP - no semi-colon will PASS
+      sql("ALTER TABLE my_table DROP CONSTRAINT aaa")
+
+      // DROP - with a semi-colon will FAIL
+      val e2 = intercept[ParseException] {
+        sql("ALTER TABLE my_table DROP CONSTRAINT bbb;")
```

Comment:
We can just change https://github.com/delta-io/delta/blob/master/core/src/main/antlr4/io/delta/sql/parser/DeltaSqlBase.g4#L70 to 
```
singleStatement
    : statement ';'? EOF
    ;
```
to fix the issue.